pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/db/ViewFilter.java

```diff
@@ -120,42 +121,30 @@ public ViewFilter(String filterName, Map<String, List<Object>> columns, String i
         this.realmId = realmId;
     }
 
-    private static SimpleResult checkUniqueFilterName(String suggestedName) {
-        SimpleResult results = inTransaction((conn) -> {
-            SimpleResult dbVals = new SimpleResult();
+    /**
+     * Returns true if a filter with the given name exists in the database,
+     * false if it does not exist.
+     */
+    public static boolean doesFilterExist(String filterName) {
+        final AtomicBoolean filterExists = new AtomicBoolean(false);
+        inTransaction((conn) -> {
             try (PreparedStatement stmt = conn.prepareStatement(SQL_CHECK_VIEW_NAME)) {
-                stmt.setString(1, suggestedName);
-                try {
-                    ResultSet rs = stmt.executeQuery();
-                    if (rs.next()) {
-                        dbVals.resultValue = "Duplicate Name";
-                    }
-                } catch (Exception e) {
-                    dbVals.resultException = e;
+                stmt.setString(1, filterName);
+                ResultSet rs = stmt.executeQuery();
+                if (rs.next()) {
+                    filterExists.set(true);
                 }
-            } catch (SQLException ex) {
-                dbVals.resultException = ex;
+            } catch (SQLException e) {
+                throw new DsmInternalError("Error checking uniqueness of filter " + filterName, e);
             }
-            return dbVals;
+            return null;
         });
-        return results;
+        return filterExists.get();
     }
 
-    public static Object saveFilter(@NonNull String filterViewToSave, String userId, @NonNull Map<String, DBElement> columnNameMap,
```

Comment:
Instead of an opaque string input that is a blob of anything, I changed it to `ViewFilter` so that the method can trust its inputs, and leave conversion from string json into the object to callers.